azure/services/aso/aso.go

```diff
@@ -252,7 +249,8 @@ func applyPatches[T deepCopier[T]](scheme *runtime.Scheme, spec azure.ASOResourc
 	if err != nil {
 		return zero, errors.Wrap(err, "failed to get GroupVersionKind for object")
 	}
-	parameters.SetGroupVersionKind(gvk)
+
+	(genruntime.MetaObject)(parameters).(interface{ SetGroupVersionKind(schema.GroupVersionKind) }).SetGroupVersionKind(gvk)
```

Comment:
A slightly nicer version of this we could do in a follow up might look like this:
```suggestion
	type gvkSetter interface{ SetGroupVersionKind(schema.GroupVersionKind) }
	var a any = parameters
	a.(gvkSetter).SetGroupVersionKind(gvk)
```